types/react-dom/test-utils/index.d.ts

```diff
@@ -298,8 +298,9 @@ export function createRenderer(): ShallowRenderer;
 declare const UNDEFINED_VOID_ONLY: unique symbol;
 // tslint:disable-next-line: void-return
 type VoidOrUndefinedOnly = void | { [UNDEFINED_VOID_ONLY]: never };
+// While act does always return Thenable, if a void function is passed, we pretend the return value is also void to not trigger dangling Promise lint rules.
+export function act(callback: () => VoidOrUndefinedOnly): void;
```

Comment:
This change breaks an earlier fix for other linting errors https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44005

Seems like there's a valid use case for both changes...

Reply:
Ooh that's a tough one. Do you have any suggestions on how to incorporate every solution into one type?

It seems like users with `strictNullChecks: false` would be a much smaller subset than those with strict. So if only one can be supported, it's best to support the strictest use-case.